rust/syntect-plugin/src/main.rs

```diff
@@ -223,10 +222,10 @@ impl<'a> Syntect<'a> {
     /// Checks if a newline has been inserted, and if so inserts whitespace
     /// as necessary.
     fn do_indentation(&mut self, view: &mut MyView, start: usize,
-                      end: usize, text: &str) -> Option<PluginEdit> {
+                      end: usize, text: &str) -> Option<u64> {
```

Comment:
ditto, since we don't need to return an edit we don't need to return anything.

Reply:
I thought that we needed to return an Int acknowledgement here, but I guess that is not the case. The plugin_lib sends an acknowledgement when plugin receives any event to the plugin. Right?

Reply:
Correct: whenever the plugin receives a new revision from core (via the `update` rpc) the plugin sends an acknowledgement back. The contract here is that the plugin will not subsequently send any edits to core that are based of revisions older than the most recently ack'd.

This is an Int for convenience, but the plugin lib can just fill that in on that side, and the return value has no meaning beyond an ack. It could just as easily be any other type.

Reply:
To clarify: because the `View` object actually constructs any edits, we know that once the plugin lib has received the update, and has updated it's internal `rev`, it will no longer be using any previous value. The actual plugin doesn't need to know any of this; it's just bookkeeping. Therefore the plugin-lib can update the plugin without needing a response; the plugin-lib will always respond to the core RPC to acknowledge, independent of any action taken by the particular plugin.